TeamCode/src/main/java/org/firstinspires/ftc/teamcode/commands/groups/ZeroRing.java

```diff
@@ -38,21 +38,23 @@ public ZeroRing(MecanumDriveSubsystem drive, WobbleSubsystem wobbleSystem, Shoot
                 .lineToConstantHeading(shootPose)
                 .build();
 
-        Vector2d secondWobble = traj2.end().vec().plus(new Vector2d(-12.0, 12.0));
+        Vector2d secondWobble = traj2.end().vec().plus(new Vector2d(-6.0, 15.5));
 
         Trajectory traj3 = drive.trajectoryBuilder(traj2.end(), 0.0)
                 .splineToLinearHeading(new Pose2d(secondWobble, 0.0), Math.toRadians(-90.0))
                 .build();
 
-        Trajectory traj4 = drive.trajectoryBuilder(traj3.end(), false)
-                .splineToConstantHeading(traj3.end().vec().plus(new Vector2d(-2.2, 8.0)), 0.0)
-                .build();
 
-        Trajectory traj5 = drive.trajectoryBuilder(traj4.end(), 0)
-                .splineTo(traj4.end().vec().plus(new Vector2d(16.0, 0.0)), 0.0)
+        Trajectory traj4 = drive.trajectoryBuilder(traj3.end(), 0)
+                .splineTo(traj3.end().vec().plus(new Vector2d(16.0, 0.0)), 0.0)
                 .splineToLinearHeading(traj1.end().plus(new Pose2d(-15.0, -12.0, Math.toRadians(180.0))), 0.0)
                 .build();
 
+        Trajectory traj5 = drive.trajectoryBuilder(traj4.end(),  Math.toRadians(180.0))
+                .splineTo(traj4.end().vec().plus(new Vector2d(0.0, 16.0)), 0.0)
+                .splineTo(traj4.end().vec().plus(new Vector2d(-16.0, 16.0)), 0.0)
+                .build();
+
         addCommands(
                 new InstantCommand(shooter::shoot, shooter),
                 new TrajectoryFollowerCommand(drive, traj0),
```

Comment:
Put this in a parallel command group with `Com_PutDown` (meaning move the Com_PutDown` up here for the first wobble).